src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -3307,7 +3321,25 @@ bool StorageReplicatedMergeTree::fetchPart(const String & part_name, const Stora
               * If you do not have time, in case of losing the session, when you restart the server - see the `ReplicatedMergeTreeRestartingThread::updateQuorumIfWeHavePart` method.
               */
             if (quorum)
-                updateQuorum(part_name);
+            {
+                /// Check if this quorum insert is parallel or not
+                if (zookeeper->exists(zookeeper_path + "/quorum/status"))
```

Comment:
.

Reply:
If there are linear and parallel quorum inserts happening concurrently (so, both nodes exist), and we are fetching part that was written with parallel quorum, the quorum will be updated incorrectly.

Looks like we just need to swap these conditions.